publisher.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+	"io/ioutil"
+	"log"
+	"github.com/jlaffaye/ftp"
+	"bytes"
+	"sync"
+	"io"
+)
+
+var lock = &sync.Mutex{}
```

Comment:
Yes otherwise all the go routines attempt to open an FTP connection at the same time and the FTP server doesn't like it as its coming from the same IP. Basically each connection needs to establish a session and immediately log in before the next connection is open. Locking on that method was the only way I could see to do it. Open to suggestions though?